### PR TITLE
EOL support for JRuby

### PR DESCRIPTION
**Pull request by basil**
The last piece of the puzzle for [JEP-7](https://github.com/jenkinsci/jep/blob/master/jep/7/README.adoc): dropping JRuby support from core. This breaks support for about two dozen infrequently used plugins, which are being suspended from distribution in jenkins-infra/update-center2#565. Since these plugins were deprecated so recently, my proposal is that we keep this PR on hold it and merge it _after_ the next LTS has shipped, at the beginning of a new LTS cycle.

### Proposed changelog entries

Support for [JRuby-based plugins](https://www.jenkins.io/blog/2021/12/22/deprecated-ruby-runtime/) has been removed.

### Proposed upgrade guidelines

Support for JRuby-based plugins has been removed. See [the blog post](https://www.jenkins.io/blog/2021/12/22/deprecated-ruby-runtime/) for migration notes.

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/resources/jenkins/security/whitelisted-classes.txt
Comment (timja) [inline core/src/main/resources/jenkins/security/whitelisted-classes.txt:147]: I've archived this repo and jenkins.rb 😄

Comment (basil) [reply core/src/main/resources/jenkins/security/whitelisted-classes.txt:147]: As of jenkins-infra/update-center2#565 these need to be archived as well:

- https://github.com/jenkinsci/buddycloud-plugin
- https://github.com/jenkinsci/capitomcat-plugin
- https://github.com/jenkinsci/chef-plugin
- https://github.com/jenkinsci/ci-skip-plugin
- https://github.com/jenkinsci/commit-message-trigger-plugin
- https://github.com/jenkinsci/gitlab-hook-plugin
- https://github.com/jenkinsci/git-notes-plugin
- https://github.com/jenkinsci/installshield-plugin
- https://github.com/jenkinsci/jenkins.py
- https://github.com/jenkinsci/pathignore-plugin
- https://github.com/jenkinsci/pyenv-plugin
- https://github.com/jenkinsci/rbenv-plugin
- https://github.com/jenkinsci/rvm-plugin
- https://github.com/jenkinsci/singleuseslave-plugin
- https://github.com/jenkinsci/travis-yml-plugin
- https://github.com/jenkinsci/yammer-plugin

Comment (timja) [reply core/src/main/resources/jenkins/security/whitelisted-classes.txt:147]: done thanks for the list

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by jglick — COMMENTED** [review]
You forgot https://github.com/jenkinsci/jenkins/blob/2e6e26c00b2b7c8f46e1cbcf26eb6749e75e2cdf/core/src/main/java/hudson/util/StreamTaskListener.java#L51

**Comment by basil** [comment]
This PR is good enough I think.

**Comment by jglick** [comment]
Originally reminded in https://github.com/jenkinsci/jep/pull/356#issuecomment-999638017. This is a piece of technical debt which was kept for the express purpose of retaining compatibility with weird JRuby infrastructure. It should be cleaned up if and when the JRuby support is removed generally, which is here. Should be simple enough; can offer an add-on commit if you like (not possible to do via GH suggestion unfortunately).

**Comment by basil** [comment]
I filed any latent curiosity about `AbstractTaskListener` for another day. Feel free to open a separate PR to clean it up.

**Pull request merged**
